### PR TITLE
Readthedocs via makefile

### PR DESCRIPTION
**Pull request by rustyrussell**
I stole @darosior #2927 and Makefilized it.  Now 'make doc-all' Does The Right Thing.

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -22,3 +22,44 @@ c-lightning Documentation
    Coding Style Guideline <STYLE.md>
    MAKING-RELEASES.md
    CHANGELOG.md
+
+.. toctree::
+   :maxdepth: 1
+   :caption: Manpages
+
+   doc/lightning-cli.1.md
```
Comment (cdecker) [inline doc/index.rst]: These paths are relative to the `index.rst` file, therefore should not have the `doc/` path.

**Comment by ZmnSCPxj** [comment]
I would mildly suggest instead:

* Find some tool to convert Markdown to manpage.
* Hand-convert (or auto-convert but with a sentient intelligence postprocessing pass) our current a2x files to Markdown.

Also suggest doing this after 0.7.2 instead.

**Comment by darosior** [comment]
> I would mildly suggest instead:
> 
> * Find some tool to convert Markdown to manpage.
> * Hand-convert (or auto-convert but with a sentient intelligence postprocessing pass) our current a2x files to Markdown.
> 
> Also suggest doing this after 0.7.2 instead.

https://pypi.org/project/mrkd/ seems able to do that. I'll try (if we end up doing it this way) it since I think it's better to use pip dep that `pandoc` (which seems able to do that, too).

**Comment by jb55** [comment]
> it's better to use pip dep than pandoc 

why's that? I've always had the best experience using pandoc for doc conversions.

**Comment by darosior** [comment]
> > it's better to use pip dep than pandoc
> 
> why's that? I've always had the best experience using pandoc for doc conversions.

Less dependencies ?

**Comment by cdecker** [comment]
> > it's better to use pip dep than pandoc
> 
> why's that? I've always had the best experience using pandoc for doc conversions.

Sorry, that's easy to mistake: I meant we shouldn't use Ubuntu packages to install python dependencies, but rather have a top-level `requirements.txt` file listing python dependencies and then using `pip` to install them into the python environment. That way we are independent of the OS package manager and can use more up to date dependencies.

**Comment by rustyrussell** [comment]
Closed in favor of #2936

**Comment by ZmnSCPxj** [comment]
Closing as per https://github.com/ElementsProject/lightning/pull/2928#issuecomment-523284962 it seems Rusty clicked wrong button with human fingers.

**Pull request closed without merging**